packages/toolpad-app/src/toolpadDataSources/rest/client.tsx

```diff
@@ -256,9 +256,11 @@ function QueryEditor({
 
   const connectionParams = isBrowserSide ? null : rawConnectionParams;
   const baseUrl = isBrowserSide ? null : connectionParams?.baseUrl ?? null;
-
+  // input.attributes.query.url will be reset when it's empty
   const urlValue: BindableAttrValue<string> =
```

Comment:
Alternatively, I believe this could be written as
```tsx
  const urlValue: BindableAttrValue<string> =
    input.attributes.query.url ?? getDefaultUrl(config, connectionParams);
```

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing